view/frontend/templates/checkout/success.phtml

```diff
@@ -45,71 +45,95 @@
+                const currency = '<?= $block->escapeJs($block->getOrder()->getOrderCurrencyCode()) ?>';
+                const locale = '<?= $block->escapeJs($block->getLocale()) ?>';
+                const environment = '<?= $block->escapeJs($block->getEnvironment()) ?>';
+                const clientKey = '<?= $block->escapeJs($block->getClientKey()) ?>';
+                const orderId = '<?= $block->escapeJs($block->getOrder()->getId()) ?>';
                 const isLoggedIn = Boolean(<?= $block->escapeJs($block->getIsCustomerLoggedIn()) ?>);
-                const maskedQuoteId = "<?= $block->escapeJs($block->getMaskedQuoteId()) ?>";
+                const maskedQuoteId = '<?= $block->escapeJs($block->getMaskedQuoteId()) ?>';
 
-                adyenPaymentService.donate(payload, isLoggedIn, orderId, maskedQuoteId)
-                    .done(function (response) {
-                        component.setStatus("success");
-                    })
-                    .fail(function (response) {
-                        component.setStatus('error');
-                    });
-            } else {
-                component.setStatus('ready');
-            }
-        }
-        function handleOnCancel(state, component) {
-            // Redirect to default 'Continue Shopping' action
-            let continueActionUrl = $('.primary.action.continue')[0].href;
-            window.location = continueActionUrl;
-        }
-        const donationConfig = {
-            amounts: {
-                currency: "<?= $block->escapeJs($block->getOrder()->getOrderCurrencyCode()) ?>",
-                values: [<?= $block->escapeJs($block->getDonationComponentConfiguration()['donationAmounts']); ?>]
-            },
-            backgroundUrl: "<?= $block->escapeJs($block->getDonationComponentConfiguration()['backgroundUrl']); ?>",
-            description: "<?= $block->escapeJs($block->getDonationComponentConfiguration()['description']); ?>",
-            logoUrl: "<?= $block->escapeJs($block->getDonationComponentConfiguration()['logoUrl']); ?>",
-            name: "<?= $block->escapeJs($block->getDonationComponentConfiguration()['name']); ?>",
-            url: "<?= $block->escapeJs($block->getDonationComponentConfiguration()['website']); ?>",
-            showCancelButton: true,
-            onDonate: handleOnDonate,
-            onCancel: handleOnCancel
-        };
-        var checkoutComponent = await AdyenCheckout({
-            locale: '<?= $block->escapeJs($block->getLocale()); ?>',
-            environment: '<?= $block->escapeJs($block->getEnvironment()); ?>',
-            clientKey: '<?= $block->escapeJs($block->getClientKey()); ?>'
+                if (!merchantAccount || !currency) {
+                    console.warn('Missing merchant account or currency for donation component.');
+                    return;
+                }
+
+                try {
+                    const payload = {
+                        merchantAccount: merchantAccount,
```

Comment:
Is this code would be evaluated in the browser or in the backend? `/donationCampaigns` call should be done on the BE and never from within a browser.
Also `/donationCampaigns` call should only be done if the previous transaction response includes `donationToken` property.